pandas/core/arrays/datetimes.py

```diff
@@ -125,17 +129,29 @@ def wrapper(self, other):
                 return ops.invalid_comparison(self, other, op)
 
             if is_object_dtype(other):
-                result = op(self.astype('O'), np.array(other))
+                # We use ops._comp_method_OBJECT_ARRAY to ensure that
+                # we raise when comparing tz and non-tz arrays
+                with np.errstate(all='ignore'):
+                    result = ops._comp_method_OBJECT_ARRAY(op,
```

Comment:
is there a test specific to this?

Reply:
Any of the arithmetic ops that assert comparing a tz-aware and non-tz-aware object raises.

I suspect separating this will be challenging since you would need to write `indexes/base.py::_make_comparison_op`.

Reply:
For future readers: tests.arithmetic.test_datetime64.test_comparison_tzawareness_compat case with `box_with_arra y = pd.Series` fails without this edit when we stop having DatetimeIndex subclass DatetimeArray.  We can recreate this in master:

```
dr = pd.date_range('2016-01-01', periods=6)
dz = dr.tz_localize('US/Pacific')

>>> dr._eadata == np.array(list(dz))   # should raise
False
```

I'll fix+test this in the upcoming branch that stops mixing DTA/TDA into DTI/TDI.